### DISCUSSION TITLE
Order of passes shown from lock screen notification

### DISCUSSION THREAD

**Discussion by natescottdavis** [alexandercerutti/passkit-generator#224] [Q&A]
I really appreciate all the work that's been done for this package, it's really helping me with my current project! While this question isn't `passkit-generator` specific, I can't seem to find any documentation about it on the web and am hopeful that someone here might know the answer...

When multiple passes are added to wallet with a .pkpasses file, the order of the passes seems to be determined by the filename of the individual .pkpass files that are zipped into the .pkpasses bundle.  This works great for me as I'd like the tickets to be arranged in a specific order. When I specify a `relevantDate` for each of the tickets, there is a single lock screen notification because the passes all have the same `groupingIdentifier`, which is also great. However, the order of the passes when they're brought up from the lock screen notification doesn't seem to be retained. From the handful of times I've tried it, it seems inconsistent. 

Does anyone know if there is way to ensure ordering for passes with the same `groupingIdentifier` that are relevant from the lock screen?

**Comment by alexandercerutti** [comment]
Hi @natescottdavis! Thanks for using it!
Sure, no problem even if the question is not strictly related to passkit-generator.

Relevance has always been a complex theme. There's never been an exact documentation about this and about when the pass become actually relevant, especially when the relevant factor is the datetime.

Another person asked me this and we weren't able to figure it out, because it is not even that exact. Might also depend on the type of the pass (if I remember correctly only `boardingPass` and `eventTicket` support it).

I don't even think there is a specific way for ensuring the order, but giving enough time-space between the tickets relevancy such that the relevant date doesn't overlap of several hours perhaps.

Furthermore, things are going to change now with iOS 18 as, if we are right, `relevantDate` might become deprecated for `relevantDates` (see iOS 18 discussion for it).

Hope this helps.

**Comment by natescottdavis** [comment]
Thanks @alexandercerutti, I appreciate it -- I'll check out the new iOS 18 stuff and then maybe file a ticket with Apple :-) Thanks again!

**Reply by alexandercerutti** [reply]
If you find something new, let me know!

In the meanwhile, if you could leave a star ⭐️ on the project, that would be great!
